src/Marten/Linq/Parsing/Methods/StringComparisonParser.cs

```diff
@@ -28,9 +28,20 @@ public ISqlFragment Parse(IFieldMapping mapping, ISerializer serializer, MethodC
         {
             var locator = GetLocator(mapping, expression);
 
-            var value = expression.Arguments.OfType<ConstantExpression>().FirstOrDefault();
+            ConstantExpression value;
+            if (expression.Object.NodeType == ExpressionType.Constant)
```

Comment:
You will need to add a null check `expression.Object?.NodeType == ExpressionType.Constant` to fix the failing tests.

Reply:
Awesome, thanks! I was going to look in to it but didn't have much time to spare.

Reply:
No worries, thanks very much for the PR and the tests 🏆